app/screens/Charts/Charts.tsx

```diff
@@ -0,0 +1,117 @@
+import React, { useEffect } from 'react';
+import { View, Text } from 'react-native';
+import { useAppDispatch, useAppSelector } from '../../hooks';
+
+import {
+  exchange18Slice,
+  exchange19Slice,
+  exchange20Slice,
+} from '../../redux/slices';
+
+import {
+  Chart,
+  Line,
+  Area,
+  HorizontalAxis,
+  VerticalAxis,
+} from 'react-native-responsive-linechart';
+
+export const Charts = () => {
+  const dispatch = useAppDispatch();
+  const data18 = useAppSelector(state => state.course18);
+  const data19 = useAppSelector(state => state.course19);
+  const data20 = useAppSelector(state => state.course20);
+
+  useEffect(() => {
+    dispatch(exchange18Slice.actions.getCourseStatistic18());
+    dispatch(exchange19Slice.actions.getCourseStatistic19());
+    dispatch(exchange20Slice.actions.getCourseStatistic20());
+  }, [dispatch]);
+
+  const usd18 = useAppSelector(
+    state => data18.course?.exchangeRate[16].saleRateNB,
+  );
+  const usd19 = useAppSelector(
+    state => data19.course?.exchangeRate[16].saleRateNB,
+  );
+  const usd20 = useAppSelector(
+    state => data20.course?.exchangeRate[23].saleRateNB,
+  );
+  const eur18 = useAppSelector(
+    state => data18.course?.exchangeRate[22].saleRateNB,
+  );
```

Comment:
it's not ok to use useAppSelector and then use variable from it again in selector  - you need to rewrite it